pmd-core/src/main/java/net/sourceforge/pmd/lang/metrics/ParameterizedMetricKey.java

```diff
@@ -57,13 +59,22 @@ private static int code(MetricKey key, MetricVersion version) {
     }
 
 
-    /** Builds a parameterized metric key. */
-    public static ParameterizedMetricKey getInstance(MetricKey<? extends Node> key, MetricVersion version) {
+    /**
+     * Builds a parameterized metric key.
+     *
+     * @param key     The key
+     * @param version The version
+     * @param <N>     The type of node of the metrickey
+     *
+     * @return An instance of parameterized metric key corresponding to the parameters
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
If this suppress is only needed for the cast in the return statement, you can limit the scope of this suppress warnings with an additional variable:

    @SuppressWarnings("unchecked")
    ParameterizedMetricKey<N> result = (ParameterizedMetricKey<N>) POOL.get(code);
    return result;

That way, not all warnings in this method are ignored, just the one we want.